cdc/sink/mysql.go

```diff
@@ -383,6 +395,28 @@ func querySQLMode(ctx context.Context, db *sql.DB) (sqlMode string, err error) {
 	return
 }
 
+// check whether the target charset is supported
+func checkCharsetSupport(ctx context.Context, db *sql.DB, charsetName string) (bool, error) {
+	// validate charsetName
+	_, err := charset.GetCharsetInfo(charsetName)
+	if err != nil {
+		return false, errors.Trace(err)
+	}
+
+	var characterSetName string
+	querySQL := fmt.Sprintf("select character_set_name from information_schema.character_sets "+
+		"where character_set_name = '%s';", charsetName)
+	err = db.QueryRowContext(ctx, querySQL).Scan(&characterSetName)
+	if err != nil && err != sql.ErrNoRows {
+		return false, cerror.WrapError(cerror.ErrMySQLQueryError, err)
+	}
+	if err != nil {
+		return false, nil
```

Comment:
This is a typo?

Reply:
> This is a typo?

For the caller of `checkCharsetSupport`, we only capture the errors other than `sql.ErrNoRows`. So if the error is `sql.ErrNoRows`, we can simply return the error as nil and check the bool value.